src/ovm/ObolValidatorManager.sol

```diff
@@ -0,0 +1,509 @@
+    consolidationSystemContract = _consolidationSystemContract;
+    withdrawalSystemContract = _withdrawalSystemContract;
+    depositSystemContract = _depositSystemContract;
+    beneficiaryRecipient = _beneficiaryRecipient;
+    rewardRecipient = _rewardRecipient;
+    principalThreshold = _principalThreshold;
+
+    _initializeOwner(_owner);
+  }
+
+  /// -----------------------------------------------------------------------
+  /// functions
+  /// -----------------------------------------------------------------------
+
+  /// -----------------------------------------------------------------------
+  /// functions - public & external
+  /// -----------------------------------------------------------------------
+
+  /// @dev Fallback function to receive ETH
+  ///      Because we do not use Clone, we must implement this explicitly
+  receive() external payable {}
+
+  /// @inheritdoc IObolValidatorManager
+  function deposit(
+    bytes calldata pubkey,
+    bytes calldata withdrawal_credentials,
+    bytes calldata signature,
+    bytes32 deposit_data_root
+  ) external payable onlyOwnerOrRoles(DEPOSIT_ROLE) {
+    uint256 oldAmountOfPrincipalStake = amountOfPrincipalStake;
+    amountOfPrincipalStake += msg.value;
+    IDepositContract(depositSystemContract).deposit{value: msg.value}(
+      pubkey,
+      withdrawal_credentials,
+      signature,
+      deposit_data_root
+    );
+
+    emit PrincipalStakeAmountUpdated(amountOfPrincipalStake, oldAmountOfPrincipalStake);
+  }
+
+  /// @inheritdoc IObolValidatorManager
+  function setBeneficiary(address newBeneficiaryRecipient) external onlyOwnerOrRoles(SET_BENEFICIARY_ROLE) {
+    if (newBeneficiaryRecipient == address(0)) {
+      revert InvalidRequest_Params();
+    }
+
+    beneficiaryRecipient = newBeneficiaryRecipient;
+
+    emit BeneficiaryUpdated(newBeneficiaryRecipient);
+  }
+
+  /// @inheritdoc IObolValidatorManager
+  function setAmountOfPrincipalStake(uint256 newAmount) external onlyOwnerOrRoles(SET_BENEFICIARY_ROLE) {
+    if (newAmount == amountOfPrincipalStake) {
+      return;
+    }
+
+    uint256 oldAmount = amountOfPrincipalStake;
+    amountOfPrincipalStake = newAmount;
+
+    emit PrincipalStakeAmountUpdated(newAmount, oldAmount);
+  }
+
+  /// @inheritdoc IObolValidatorManager
+  function setRewardRecipient(address newRewardRecipient) external onlyOwnerOrRoles(SET_REWARD_ROLE) {
+    if (newRewardRecipient == address(0)) {
+      revert InvalidRequest_Params();
+    }
+
+    rewardRecipient = newRewardRecipient;
+
+    emit RewardRecipientUpdated(newRewardRecipient);
+  }
+
+  /// @inheritdoc IObolValidatorManager
+  function sweep(address beneficiary, uint256 amount) external nonReentrant {
+    address recipient = beneficiaryRecipient;
+    if (beneficiary != address(0)) {
+      _checkOwner();
+      recipient = beneficiary;
+    }
+
+    // If amount is zero, sweep all funds on the contract
+    uint256 sweepAmount = amount == 0 ? address(this).balance : amount;
```

Comment:
wouldn't this take rewards that should go to the reward recipient in our design? i.e. we should be checking like `fundsPendingWithdrawal` or `pullBalances[beneficiary]` or something not address(this).balance? 

I feel like this method should just be a wrapper around our `withdrawPullBalance()` method? we might change the name and function selector for that method to include amount though, as we've had to change the name already because of the collision with `withdraw()`. 

lets maybe sync on what we do for this behaviour, because i think its a bit messy right now, and we're probably better off streamlining it.